src/terms/local.jl

```diff
@@ -69,17 +69,29 @@ Atomic local potential defined by `model.atoms`.
 """
 struct AtomicLocal end
 function (::AtomicLocal)(basis::PlaneWaveBasis{T}) where {T}
-    model = basis.model
     # pot_fourier is <e_G|V|e_G'> expanded in a basis of e_{G-G'}
     # Since V is a sum of radial functions located at atomic
     # positions, this involves a form factor (`local_potential_fourier`)
     # and a structure factor e^{-i G·r}
+    model = basis.model
+    G_cart = G_vectors_cart(basis)
+
+    form_factors = IdDict{Tuple{Int,T},T}()
```

Comment:
clarify that this is an optimization and why it's needed

Reply:
Might as well have an outer loop over groups, simplifies the logic instead of passing through the groups twice

Reply:
I thought about it, but with the constraint of the map syntax, it's not very doable because each entry in the fourier potential is a sum over groups. It's doable with an array I can index into, but I'm not sure whether performance is better looping over small arrays many times or a massive array a few times.

Reply:
Are you sure? How about having a sub-function to compute the contribution of a single group and then just have a simple `sum(pot_fourier_of_group, groups)` (where `pot_fourier_of_group` of course is a terrible name)?

Reply:
Or, you know, just pot_fourier = zero(), for group in groups pot_fourier += ..., end. These functional thingies are nice to play with, but sometimes nothing beats plain old imperative loops :-)

Reply:
Michael asked me to keep the map syntax here for AD/GPU. I could do `sum(groups) do map(G_vecs) do ... end end`, but I think this would allocate `N_groups` copies of `pot_fourier`, which are quite big arrays.